SpringKonstant/src/org/usfirst/frc6357/robotcode/subsystems/TFMini.java

```diff
@@ -204,25 +170,65 @@ private void runLiDAR()
             {
                while(startRead)
                {
-            	   if(amountOfBytes() >= 9)
-            	   {
-            		   byte bytes[] = readBytes();
-            		   if(isValid(bytes))
-            		   {
-            			   distL = bytes[2];
-            			   distH = bytes[3];
-            			   strengthL = bytes[4];
-            			   strengthH = bytes[5];
-            			   reservedBytes = bytes[6];
-            			   signalQualityDegree = bytes[7];
-            		   }
-            	   }
+                   byte bytes[] = sensor.read(1);
+                   if(bytes.length > 0)
+                   {
+                       switch(state)
+                       {
+                           case WAIT_START1:
+                               if(bytes[0] == 0x59)
+                               {
+                                   count = 0;
+                                   checksum = bytes[0];
+                                   packet[0] = bytes[0];
+                                   state = ReceivingState.WAIT_START2;
+                               }
+                               break;
+                           case WAIT_START2:
```

Comment:
This WAIT_START2 state appears to be inspecting the same location in the packet as WAIT_START1. Or is the intent that we only fetch one byte at a time?